src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -995,6 +974,61 @@ private BoundCollectionExpression ConvertCollectionExpression(
                     _ = GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_List_T__ToArray, diagnostics, syntax: syntax);
                 }
 
+                if (collectionTypeKind is CollectionExpressionTypeKind.CollectionBuilder)
+                {
+                    Debug.Assert(elementType is { });
+
+                    ((NamedTypeSymbol)targetType).HasCollectionBuilderAttribute(out TypeSymbol? builderType, out string? methodName);
+
+                    var collectionBuilderCandidates = GetAndValidateCollectionBuilderMethods(syntax, ((NamedTypeSymbol)targetType).OriginalDefinition, builderType, methodName, diagnostics);
+                    if (collectionBuilderCandidates.IsEmpty)
+                    {
+                        return BindCollectionExpressionForErrorRecovery(node, targetType, inConversion: true, diagnostics);
+                    }
+
+                    Debug.Assert(builderType is { });
+                    Debug.Assert(methodName is { });
```

Comment:
nit: consider using the same assertion as you had elsewhere (`Debug.Assert(!string.IsNullOrEmpty(methodName));`)